mapreduce/src/main/java/org/infinispan/hadoop/mapreduce/impl/InfinispanRecordWriter.java

```diff
@@ -0,0 +1,37 @@
+package org.infinispan.hadoop.mapreduce.impl;
+
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.hadoop.mapreduce.KeyValueConverter;
+
+import java.io.IOException;
+
+/**
+ * @param <K>
+ * @param <V>
+ * @author gustavonalle
+ * @since 0.1
+ */
+public class InfinispanRecordWriter<K, V> extends RecordWriter<K, V> {
```

Comment:
Nice idea. I did something similar to Spark with a WRITE_BATCH config and putAll calls, but I did not use the segment for each key to create the batches: for that it's only a matter of exposing the ```getKeySegments``` in the ```cache.getTopologyInfo```